Libraries/LibWeb/CSS/Parser/PropertyParsing.cpp

```diff
@@ -1658,6 +1662,120 @@ RefPtr<CSSStyleValue const> Parser::parse_border_value(PropertyID property_id, T
         { border_width.release_nonnull(), border_style.release_nonnull(), border_color.release_nonnull() });
 }
 
+RefPtr<CSSStyleValue const> Parser::parse_border_image_value(TokenStream<ComponentValue>& tokens)
+{
+    // <'border-image-source'> || <'border-image-slice'> [ / <'border-image-width'> | / <'border-image-width'>? / <'border-image-outset'> ]? || <'border-image-repeat'>
```

Comment:
I think you've misinterpreted this syntax (though that's fair, it's a bit confusing). I was about to submit a patch but then noticed this technically isn't ambiguous - `||` has lower precedence than the other operators here.

I think it's meant to be 1+ of these, in any order:
- `<'border-image-source'>`
- `<'border-image-slice'> [ / <'border-image-width'> | / <'border-image-width'>? / <'border-image-outset'> ]?`
- `<'border-image-repeat'>`

But I think what you're parsing is this, in order:
1. `<'border-image-source'> || <'border-image-slice'>`
2. `[ / <'border-image-width'> | / <'border-image-width'>? / <'border-image-outset'> ]?`
3. `<'border-image-repeat'>`

This is probably the most unpleasant property to parse that I've ever seen. :melting_face:

Reply:
I've now rewritten it in a way that makes it a bit clearer what is being parsed when.